### DISCUSSION TITLE
InitialRoute

### DISCUSSION THREAD

**Discussion by andim27** [nylo-core/nylo#62] [General] [answered]
InitialRoute is useful ability.But if we use authPage:true in the real app we have 2 initial routes.And first route always located in the stack tree
Example:
 router.route(HomePage.path, (context) => HomePage(), initialRoute: true);
 router.route(AuthPage.path, (context) => AuthPage(), authPage: true);
If we open app with Auth ability the HomePage presents in the widgets tree.According to the building logics in this case we have main initial route:AuthPage. But we can't define variable  initialRoute: true in the AuthPage. It's bad.How to get rid of two initial route-page?

**Comment by agordn52** [comment, marked as answer]
Hi @andim27,

It's to do with Flutter's routing strategy, you can learn more from this StackOverflow link.
https://stackoverflow.com/questions/56468319/navigator-always-start-with-route

Make sure that the **HomePage** `path` is not "/".
You should update the path to "/home-page"

**Comment by andim27** [comment]
ok
